src/Octoshift/GithubClient.cs

```diff
@@ -156,12 +162,24 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url, Dictionary
             {
                 response.EnsureSuccessStatusCode();
             }
-            else if (response.StatusCode != status)
+            else if (response.StatusCode != status && RateLimitRemaining(headers) != 0)
             {
                 throw new HttpRequestException($"Expected status code {status} but got {response.StatusCode}", null, response.StatusCode);
             }
 
-            return (content, response.Headers.ToArray());
+            CheckForRetryDelay(headers);
+
+            return (content, headers);
+        }
+
+        private async Task ApplyRetryDelayAsync()
+        {
+            if (_retryDelay > 0.0)
+            {
+                _log.LogWarning($"THROTTLING IN EFFECT. Waiting {(int)_retryDelay} ms");
```

Comment:
I would show the wait time in seconds to be more user-friendly (I even considered suggesting minutes, but seconds probably works).  I would also reword this a bit and not use the word throttling here, because the way github does things (unlike ADO) I don't really think of it as throttling.  Maybe something like "GitHub rate limit exceeded. Waiting 329 seconds before continuing".